Providers/RustSyncManager.swift

```diff
@@ -254,7 +254,18 @@ public class RustSyncManager: NSObject, SyncManager {
         return clearPrefs()
     }
 
-    public func getEngineEnablementChangesForAccount() -> [String: Bool] {
+    public func checkCreditCardEngineEnablement() -> Bool {
+        var engineEnabled = false
+        guard let declined = UserDefaults.standard.stringArray(forKey: fxaDeclinedEngines) else {
+            let engine = "creditcards"
+            engineEnabled = prefsForSync.boolForKey("engine.\(engine).enabled") ?? false
+            return engineEnabled
+        }
+        UserDefaults.standard.removeObject(forKey: fxaDeclinedEngines)
+        return engineEnabled
+    }
```

Comment:
FYI: @lougeniaC64 added this change for checking telemetry

Reply:
@nbhasin2 I don't think this is right. For one, I don't think you want to call `UserDefaults.standard.removeObject(forKey: fxaDeclinedEngines)` because it might interfere with the user declining an engine being recognized by sync.

Also, after the guard clause on line 263 you're going to want to check that the credit card engine is included in `declined`. If it is, the engine is disabled, otherwise it's enabled.

Reply:
Thank you for the miss on my end @lougeniaC64, fixed :)